tests/5.0/flush/test_flush_no_memory_order_clause.c

```diff
@@ -0,0 +1,41 @@
+//===---- test_flush_no_memory_order_clause.c -------------------------------===//
+// 
+// OpenMP API Version 5.0 
+// 
+// This is a test of the flush directive with no memory-order-clause specified.
+// Additionally, atomic constructs are used alongside the explicit flush directives
+// to specify memory ordering amongst the two threads.
+//
+// Based on OpenMP 5.0 Example aquire_release.3.c
+//===------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+int errors = 0;
+
+int main() {
+
+   int x = 0, y = 0;
+   #pragma omp parallel num_threads(2)
+   {
+      int thrd = omp_get_thread_num();
+      if (thrd == 0) {
+         x = 10;
+         #pragma omp flush
+         #pragma omp atomic write
+         y = 1;
+      } else {
+         int tmp = 0;
+         while (tmp == 0) {
+            #pragma omp atomic read
+	    tmp = y;
+         }
+         #pragma omp flush
```

Comment:
why is this flush needed ?

Reply:
I don't think it is needed, as I understand the test.

Reply:
Actually, I think it is needed since thread 1 is reading x after thread 0 writes to x and flushes. So thread 1 needs to do an acquire flush here to ensure it has the latest version of x when x's value is checked.

Reply:
Right, the binding is to the calling thread only.